spec/requests/posts_request_spec.rb

```diff
@@ -0,0 +1,39 @@
+require 'rails_helper'
+
+RSpec.describe 'Controllers', type: :request do
+  describe 'PostsController' do
+    describe '#index' do
+      it 'returns a successful response' do
+        get '/users/1/posts'
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'renders the index template' do
+        get '/users/1/posts'
+        expect(response).to render_template(:index)
+      end
+
+      it 'includes correct placeholder text in the response body' do
+        get '/users/1/posts'
+        expect(response.body).to include('List of all posts and comment of a user')
+      end
+    end
+
+    describe '#show' do
+      it 'returns a successful response' do
+        get '/users/1/posts/4'
+        expect(response).to have_http_status(:success)
+      end
+
+      it 'renders the show template' do
+        get '/users/1/posts/2'
+        expect(response).to render_template(:show)
+      end
+
+      it 'includes correct placeholder text in the response body' do
+        get '/users/1/posts/4'
+        expect(response.body).to include('>Details of a selected post')
+      end
+    end
+  end
+end
```

Comment:
Hi @tresorsawasawa, This is confusing for me because all tests are passing when I run them on my side, to test I even put a wrong post id, and it returned a failed test, then when I put the right one as the one I have now, all tests pass. I am really having a hard time with this as you can see with the previous reviewer, 3 tests were failing in the posts controller and none in the users' controller, without changing anything the users' controller tests are failing when you run the tests. Is it possible that the error could be on your side, just checking because I'm at a loss now.

Reply:
Hi @SaneMyburg 

Can try to delete your old database, then create a new one and run migrations and see the result i.e.
`rails db:drop db:create db:migrate`
Then you can run your tests, and see the result. Even the previous reviewer, got the same result like mine